mlflow/tracing/fluent.py

```diff
@@ -260,7 +279,184 @@ def pagination_wrapper_func(number_to_get, next_page_token):
 
     get_display_handler().display_traces(results)
 
-    return results
+    traces_df = traces_to_df(results)
+    if extract_fields:
+        traces_df = extract(
+            traces=traces_df,
+            fields=extract_fields,
+            col_name=SPANS_COLUMN_NAME,
+        )
+
+    return traces_df
+
+
+def extract(
+    traces: Union[List["mlflow.entities.Trace"], "pandas.DataFrame"],
+    fields: List[str],
+    col_name: Optional[str] = None,
+) -> "pandas.DataFrame":
+    """
+    Extracts the specified fields from the spans contained in the specified traces.
+
+    Args:
+        traces: A list of MLflow Traces or a pandas DataFrame containing traces.
```

Comment:
```suggestion
        traces: A list of :py:class:`mlflow.entities.Trace` or a pandas DataFrame containing traces.
```

nit: linking to the class doc just for user convenience